tests/util_test.go

```diff
@@ -55,8 +57,80 @@ func (c *ClientTests) SetupTest() {
 		return cassette.DefaultMatcher(r, i) && (b.String() == "" || b.String() == i.Body)
 	})
 
-	r.AddFilter(func(i *cassette.Interaction) error {
-		delete(i.Request.Headers, "Authorization")
+	// Filter sensitive data from cassettes
+	// Replace value has to be type specific to its corresponding struct
+	responseBodyScrubbers := map[string]interface{}{
+		"api_keys":         []string{},
+		"children":         []string{},
+		"client_ip":        "REDACTED",
+		"test_credentials": map[string]string{},
+		"credentials":      map[string]string{},
+		"email":            "REDACTED",
+		"keys":             []*easypost.APIKey{},
+		"key":              "REDACTED",
+		"phone_number":     "REDACTED",
+		"phone":            "REDACTED",
+	}
+	r.AddSaveFilter(func(i *cassette.Interaction) error {
+		// Filter headers
+		header := i.Request.Headers
+		fmt.Print(header)
+		if i.Request.Headers["Authorization"] != nil {
+			i.Request.Headers["Authorization"] = []string{"REDACTED"}
+		}
+
+		if i.Request.Headers["User-Agent"] != nil {
+			i.Request.Headers["User-Agent"] = []string{"REDACTED"}
+		}
+
+		if i.Request.Headers["X-Client-User-Agent"] != nil {
+			i.Request.Headers["X-Client-User-Agent"] = []string{"REDACTED"}
+		}
+
+		// Filter response bodies
+		var responseBodyBytes []byte
+		var responseBodyString string
+
+		for scrubber, replaceValue := range responseBodyScrubbers {
+			if strings.Contains(i.Response.Body, scrubber) {
+				var responseBody = i.Response.Body
+				var arrayObjects []map[string]interface{}
+
+				if json.Unmarshal([]byte(responseBody), &arrayObjects) == nil {
+					err := json.Unmarshal([]byte(responseBody), &arrayObjects)
+					if err != nil {
+						fmt.Print(err)
+						os.Exit(1)
+					}
+					for index, object := range arrayObjects {
+						for key := range object {
+							if key == scrubber {
+								arrayObjects[index][key] = replaceValue
+							}
+						}
+					}
+
+					responseBodyBytes, _ = json.Marshal(arrayObjects)
+				} else {
+					responseMap := make(map[string]interface{})
+
+					err := json.Unmarshal([]byte(responseBody), &responseMap)
+					if err != nil {
+						fmt.Print(err)
+						os.Exit(1)
+					}
+
+					if _, exists := responseMap[scrubber]; exists {
```

Comment:
Explicit over implicit. Your future self (and other engineers) will thank you!

```suggestion
					if _, keyExists := responseMap[scrubber]; exists {
```